pysmt/test/examples.py

```diff
@@ -306,6 +307,100 @@ def get_example_formulae(environment=None):
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # (bv16 comp bv16) = bv0
+            Example(expr=Equals(BVComp(bv16, bv16), BVZero(1)),
+                    is_valid=False,
+                    is_sat=False,
+                    logic=pysmt.logics.QF_BV),
+
+            # (bv16 s< bv16)
+            Example(expr=BVSLT(bv16, bv16),
+                    is_valid=False,
+                    is_sat=False,
+                    logic=pysmt.logics.QF_BV),
+
+            # (bv16 s< 0_16)
+            Example(expr=BVSLT(bv16, BVZero(16)),
+                    is_valid=False,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # (bv16 u< bv16)
+            Example(expr=BVULT(bv16, bv16),
+                    is_valid=False,
+                    is_sat=False,
+                    logic=pysmt.logics.QF_BV),
+
+            # (bv16 s< 0_16)
+            Example(expr=BVULT(bv16, BVZero(16)),
+                    is_valid=False,
+                    is_sat=False,
+                    logic=pysmt.logics.QF_BV),
+
+            # 0_16 s< bv16 & ((bv16 s/ -1) s< 0)
+            Example(expr=And(BVSLT(BVZero(16), bv16),
+                             BVSLT(BVSDiv(bv16, BVNeg(BVOne(16))), BVZero(16))),
+                    is_valid=False,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # 0_16 s< bv16 & ((bv16 s% -1) s< 0)
+            Example(expr=And(BVSLT(BVZero(16), bv16),
+                             BVSLT(BVSRem(bv16, BVOne(16)), BVZero(16))),
+                    is_valid=False,
+                    is_sat=False,
+                    logic=pysmt.logics.QF_BV),
+
+            # bv16 u% 1 = 0_16
+            Example(expr=Equals(BVURem(bv16, BVOne(16)), BVZero(16)),
+                    is_valid=True,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # bv16 s% 1 = 0_16
+            Example(expr=Equals(BVSRem(bv16, BVOne(16)), BVZero(16)),
+                    is_valid=True,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # bv16 s% -1 = 0_16
+            Example(expr=Equals(BVSRem(bv16, BVNeg(BVOne(16))), BVZero(16)),
+                    is_valid=True,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # bv16 a>> 0 = bv16
+            Example(expr=Equals(BVAShr(bv16, BVZero(16)), bv16),
+                    is_valid=True,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
+            # 0 s< bv16 & bv16 a>> 1 = bv16 >> 1
+            Example(expr=And(BVSLT(BVZero(16), bv16),
+                             Equals(BVAShr(bv16, BVOne(16)),
+                                    BVLShr(bv16, BVOne(16)))),
+                    is_valid=False,
+                    is_sat=True,
+                    logic=pysmt.logics.QF_BV),
+
```

Comment:
This is an interesting point. MathSAT takes the position of forbidding negative-valued BV constants.
I think it is better to avoid overloding the ```BV(val, width)``` shortcut allowing negative values.
Consider the following case:
* ```BV(3, 2)``` This is ok and results in "11" if the bitvecor is considered unsigned, but it is an overflow error if we consider it signed

We could either add a parameter to ```BV()``` or (better) create a new shortcut ```SignedBV()```. Note that we can already create negative numbers by simply negating a positive value: ```BVNeg(BV(1,5))```.